src/sfptpd_clock.c

```diff
@@ -2484,4 +2571,163 @@ bool sfptpd_clock_is_active(const struct sfptpd_clock *clock)
 	return clock && !clock->deleted;
 }
 
+int sfptpd_clock_deduplicate(void)
+{
+	struct sfptpd_clock **snapshot;
+	struct sfptpd_clock *sys;
+	size_t count = 0;
+	/* A large settlement period is required; at least >1s in case of PPS
+	 * phc diff method being in use. */
+	const useconds_t settle = 1000000;
+	struct sfptpd_timespec test_adjustment;
+	struct sfptpd_timespec restore_adjustment;
+	struct sfptpd_timespec threshold;
+	struct sfptpd_timespec t;
+	struct sfptpd_clock_calibration *calibrations;
+	int i, j;
+	int rc;
+	int changes = 0;
+
+	clock_lock();
+
+	snapshot = clock_snapshot(&count, true);
+
+	if (count < 2)
+		goto finish;
+
+	NOTICE("clock: identifying duplicate phc clocks, a destructive operation\n");
```

Comment:
*boom* go your NICs?
If we're actually playing with time; suggestion - should we interlock with the options users use to say "Never step, only slew" and if they set both, output an error to say that're not compatible with dedupe?

Reply:
Interplay with `clock_control` sounds like a nightmare...

> boom go your NICs?

Is the concept of a _destructive operation_ not well understood in our users' circles?